lib/rubocop/cop/performance/method_object_as_block.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop identifies places where methods are converted to blocks, with the
+      # use of `&method`, and passed as arguments to `Enumerable` method calls.
+      # It is faster to replace those with explicit blocks, calling those methods inside.
+      #
+      # @example
+      #   # bad
+      #   array.map(&method(:do_something))
+      #
+      #   # good
+      #   array.map { |x| do_something(x) }
+      #
+      class MethodObjectAsBlock < Base
+        MSG = 'Use block explicitly instead of passing a method object.'
+
+        def_node_matcher :method_object?, <<~PATTERN
+          (block-pass (send nil? :method sym))
```

Comment:
I'd switch `nil?` => `_`, and test it / document it.

Reply:
You mean, that we should also detect cases when `method` also has a receiver, like in this
```ruby
out = File.open('test.txt', 'w')
[ 1, 2, 3 ].each(&out.method(:puts))
```
?

Reply:
Yes, that should be `[ 1, 2, 3 ].each{ |x| out.puts(x) }`.